ra-optimisation/rao-commons/src/main/java/com/farao_community/farao/rao_commons/linear_optimisation/fillers/OperatorsNotToOptimizeFiller.java

```diff
@@ -0,0 +1,203 @@
+        BranchCnec mostLimitingElement = RaoUtil.getMostLimitingElement(raoData.getCnecs(), preOptimVariantId, objFunction.getUnit(), false);
+        double preOptimVariantWorstMargin = RaoUtil.computeCnecMargin(mostLimitingElement, preOptimVariantId, objFunction.getUnit(), false);
+
+        String prePerimeterVariantId = raoData.getCrac().getExtension(ResultVariantManager.class).getPrePerimeterVariantId();
+        mostLimitingElement = RaoUtil.getMostLimitingElement(raoData.getCnecs(), prePerimeterVariantId, objFunction.getUnit(), false);
+        double prePerimeterVariantWorstMargin = RaoUtil.computeCnecMargin(mostLimitingElement, prePerimeterVariantId, objFunction.getUnit(), false);
+
+        return Math.min(preOptimVariantWorstMargin, prePerimeterVariantWorstMargin);
+    }
+
+    /**
+     * For CNECs of operators not sharing RAs, deactivate their participation in the definition of the minimum margin
+     * if their margin is not decreased (ie margin_decrease = 0)
+     * Do this by adding (1 - margin_decrease) * bigM to the right side of the inequality
+     * bigM is computed as 2 times the largest absolute threshold between all CNECs
+     * Of course this can be restrictive as CNECs can have hypothetically infinite margins if they are monitored in one direction only
+     * But we'll suppose for now that the minimum margin can never be greater than 1 * the largest threshold
+     */
+    private void updateMinimumMarginConstraints(RaoData raoData, LinearProblem linearProblem) {
+        MPVariable minimumMarginVariable = linearProblem.getMinimumMarginVariable();
+        if (minimumMarginVariable == null) {
+            throw new FaraoException("Minimum margin variable has not yet been created");
+        }
+
+        double bigM = 2 * getLargestCnecThreshold(raoData);
+        getCnecsForOperatorsNotToOptimize(raoData).forEach(cnec -> {
+            MPVariable marginDecreaseBinaryVariable = linearProblem.getMarginDecreaseBinaryVariable(cnec);
+            if (marginDecreaseBinaryVariable == null) {
+                throw new FaraoException(String.format("Margin decrease binary variable has not yet been created for Cnec %s", cnec.getId()));
+            }
+            updateMinimumMarginConstraint(linearProblem.getMinimumMarginConstraint(cnec, LinearProblem.MarginExtension.BELOW_THRESHOLD), marginDecreaseBinaryVariable, bigM);
+            updateMinimumMarginConstraint(linearProblem.getMinimumMarginConstraint(cnec, LinearProblem.MarginExtension.ABOVE_THRESHOLD), marginDecreaseBinaryVariable, bigM);
+            updateMinimumMarginConstraint(linearProblem.getMinimumRelativeMarginConstraint(cnec, LinearProblem.MarginExtension.BELOW_THRESHOLD), marginDecreaseBinaryVariable, bigM);
+            updateMinimumMarginConstraint(linearProblem.getMinimumRelativeMarginConstraint(cnec, LinearProblem.MarginExtension.ABOVE_THRESHOLD), marginDecreaseBinaryVariable, bigM);
```

Comment:
don't we need to check the config first to avoid a crash here if the objective function is the "normal margin" ?

Reply:
the updateMinimumMarginConstraint checks for the existence of the relative margins variable

Reply:
this way this filler adapts automatically, no need th inject the relative margins parameter

Reply:
:+1: